### DISCUSSION TITLE
refget: Get metadata alone, without sequence

### DISCUSSION THREAD

**Discussion by juliangehring** [zaeleus/noodles#178] [Q&A] [answered]
In noodles-refget, is it currently possible to pull only the metadata of a sequence, without having to have to get the sequence first?  This might be especially useful when the sequences are large and take a long time to pull, only to be ignored.

This is my current approach:

```rust
let client = Client::new(base_url);
let sequence = client.sequence(id).send().await?;
let metdata = sequence.metadata().await?;
```

Maybe there is a smarter, more efficient way of doing it with the current API that I'm missing. Any ideas are welcome :)

**Comment by zaeleus** [comment, marked as answer]
There is currently no way to get the metadata before the sequence. One workaround is to first request a single base, e.g., using the interval `1-1`, which would avoid downloading large sequences.

```rust
let sequence = client
    .sequence(id)
    .set_interval(Position::MIN..=Position::MIN)
    .send()
    .await?;
let metadata = client.metadata().await?;
```

I'll track adding a metadata request builder in #180.

**Reply by juliangehring** [reply]
Thank you for the explanation and the suggested workaround! A separate metadata builder could be a useful addition for the longer term.

**Comment by zaeleus** [comment]
1b5bb252b8f5ece0bd913e7d8031d0bab9af2dc3 adds `refget::Client::sequence_metadata`, which allows getting the metadata without first requesting the sequence, e.g.,

```rust
let metadata = client.sequence_metadata(id).send().await?;
```
